code/datums/emote_system/emote.dm

```diff
@@ -0,0 +1,223 @@
+/*******************************************************************************************************
+												Emotes
+Remember, only 1 instance of each emote is made, and used by all mobs that can access it. As such, the
+ONLY place you can safely set object variables during runtime is in New(). Otherwise override the emote.
+The only exception to this is custom emotes, which are created as needed and only last until they are finished
+VampyrBytes
+
+*******************************************************************************************************/
+
+/datum/emote
+	var/name
+	var/list/commands[0]  // commands that trigger the emote. Set these in New()
+	var/text
+	var/startText		// if you need to put something in before [user]. Should end with a space
+	var/mimeText
+	var/sound
+	var/vol = 50
+	var/audible
+	var/muzzledNoise	// if the emote is audible and you're muzzled, this is what type of noise you make (eg weak, loud). End with a space
+	var/restrained		// 1 if being restrained prevents this emote
+	var/cooldown
+
+	var/canTarget		// 1 if the emote accepts a target			Emotes only recieve 1 parameter, so its either or with these 2
+	var/takesParam		// 1 if the emote uses a non target parameter
+
+	var/targetMob		// 0 if target can be any atom, 1 if it has to be a mob
+	var/targetText = "at" // what goes inbetween user and target
+	var/spanClass = "notice"
+	var/baseLevel = 1
+	var/baseSet
+
+
+/datum/emote/New()
+	var/pathString = "[type]"
+	var/count
+	for(var/i = 1, i<=lentext(pathString), i++)
```

Comment:
lol, yeah it is. baseLevel is used by the search and needs to be set to 1 for the top level of an emote (eg /datum/emote/scream) but 0 for any that override it (eg /datum/emote/scream/human). 
Best way of automating it I could come up with was to count the / in the path. As not setting it properly would lead to some funky search results I didn't want to leave it to whoever overrides an emote remembering to set it.